libs/vkd3d/command.c

```diff
@@ -14684,8 +15023,104 @@ static void STDMETHODCALLTYPE d3d12_command_list_BeginRenderPass(d3d12_command_l
         return;
     }
 
+    d3d12_command_list_invalidate_rendering_info(list);
+    d3d12_command_list_end_current_render_pass(list, false);
+
+    prev_dsv_format = list->dsv.format ? list->dsv.format->vk_format : VK_FORMAT_UNDEFINED;
+
     list->is_inside_render_pass = true;
     list->render_pass_flags = flags;
+    memset(list->rtvs, 0, sizeof(list->rtvs));
+    memset(&list->dsv, 0, sizeof(list->dsv));
+
+    d3d12_command_list_debug_mark_begin_region(list, "BeginRenderPass");
+
+    for (i = 0, rt_index = 0; i < rt_count; i++)
+    {
+        const D3D12_RENDER_PASS_RENDER_TARGET_DESC *rt = &render_targets[i];
+
+        if (d3d12_render_pass_beginning_access_binds_to_rasterizer(rt->BeginningAccess.Type, flags, VK_IMAGE_ASPECT_COLOR_BIT))
+        {
+            if (rt_index >= ARRAY_SIZE(list->rtvs))
+            {
+                WARN("Render target count %u > %zu, ignoring extra descriptors.\n", rt_index, ARRAY_SIZE(list->rtvs));
+                continue;
+            }
+
+            if ((rtv_desc = d3d12_rtv_desc_from_cpu_handle(rt->cpuDescriptor)) && rtv_desc->resource)
+            {
+                list->rtvs[rt_index] = *rtv_desc;
+
+                VKD3D_BREADCRUMB_AUX64(rtv_desc->view->cookie);
+                VKD3D_BREADCRUMB_AUX32(i);
+                VKD3D_BREADCRUMB_TAG("RTV bind");
+
+                if (!(flags & D3D12_RENDER_PASS_FLAG_RESUMING_PASS))
+                    d3d12_command_list_load_render_pass_rtv(list, &list->rtvs[rt_index], rt);
```

Comment:
I think we're missing image initialization tracking here. If CLEAR op is used, we need to do similar things that OMSetRenderTargets does and set d3d12_command_list_track_resource_usage(false) if appropriate.

If we don't do anything, e.g. partial CLEAR or LOAD, we'll hit the init in d3d12_command_list_emit_render_pass_transition.

Reply:
Right, put the relevant calls in `d3d12_command_list_load_render_pass_rtv` for simplicity.